newIDE/app/src/AssetStore/AssetPackDialog.js

```diff
@@ -0,0 +1,210 @@
+    flex: 1,
+  },
+};
+
+type Props = {|
+  assetPack: AssetPack,
+  assetShortHeaders: Array<AssetShortHeader>,
+  addedAssetIds: Array<string>,
+  onClose: () => void,
+  onAssetPackAdded: () => void,
+  project: gdProject,
+  objectsContainer: gdObjectsContainer,
+  events: gdEventsList,
+  onObjectAddedFromAsset: (object: gdObject) => void,
+|};
+
+export const AssetPackDialog = ({
+  assetPack,
+  assetShortHeaders,
+  addedAssetIds,
+  onClose,
+  onAssetPackAdded,
+  project,
+  objectsContainer,
+  events,
+  onObjectAddedFromAsset,
+}: Props) => {
+  const missingAssetShortHeaders = assetShortHeaders.filter(
+    assetShortHeader => !addedAssetIds.includes(assetShortHeader.id)
+  );
+  const allAssetsInstalled = missingAssetShortHeaders.length === 0;
+  const noAssetsInstalled =
+    !allAssetsInstalled &&
+    missingAssetShortHeaders.length === assetShortHeaders.length;
+
+  const resourcesFetcher = useResourceFetcher();
+  const [
+    isAssetPackBeingInstalled,
+    setIsAssetPackBeingInstalled,
+  ] = React.useState<boolean>(false);
+
+  const eventsFunctionsExtensionsState = React.useContext(
+    EventsFunctionsExtensionsContext
+  );
+
+  const onInstallAssetPack = React.useCallback(
+    async (assetShortHeaders: Array<AssetShortHeader>) => {
+      if (!assetShortHeaders || !assetShortHeaders.length) return;
+      setIsAssetPackBeingInstalled(true);
+      try {
+        const installOutputs = await Promise.all(
+          assetShortHeaders.map(assetShortHeader =>
+            installAsset({
+              assetShortHeader,
+              eventsFunctionsExtensionsState,
+              project,
+              objectsContainer,
+              events,
+            })
+          )
+        );
+        installOutputs.forEach(installOutput => {
+          installOutput.createdObjects.forEach(object => {
+            onObjectAddedFromAsset(object);
+          });
+        });
+
+        await resourcesFetcher.ensureResourcesAreFetched(project);
+
+        setIsAssetPackBeingInstalled(false);
+        onAssetPackAdded();
+      } catch (error) {
+        setIsAssetPackBeingInstalled(false);
+        console.error('Error while installing the asset pack', error);
+        showErrorBox({
+          message:
+            'There was an error while installing the asset pack. Verify your internet connection or try again later.',
+          rawError: error,
+          errorId: 'install-asset-pack-error',
+        });
+      }
+    },
+    [
+      resourcesFetcher,
+      eventsFunctionsExtensionsState,
+      project,
+      objectsContainer,
+      events,
+      onObjectAddedFromAsset,
+      onAssetPackAdded,
+    ]
+  );
+
+  return (
+    <Dialog
+      title={<Trans>{assetPack.name}</Trans>}
+      open
+      cannotBeDismissed
+      actions={[
+        <FlatButton
+          key="cancel"
+          label={<Trans>Cancel</Trans>}
+          disabled={isAssetPackBeingInstalled}
+          onClick={onClose}
+        />,
```

Comment:
Does it cancel the asset downloading or just close the dialog and continue to silently download assets?
```
onClose={() => setIsAssetPackDialogInstallOpen(false)}
```